TFNetworkLayer.py

```diff
@@ -3520,21 +3520,34 @@ def get_out_data_from_opts(cls, name, sources, axes=None, axis=None, keep_dims=F
     axes = cls.get_axes(axis=axes, input_data=x)
     y_shape = list(x.batch_shape)
     out_batch_dim_axis = x.batch_dim_axis
+    out_feature_dim_axis = x.feature_dim_axis_or_unspecified
+    out_time_dim_axis = x.time_dim_axis
     if keep_dims:
       for i in axes:
         y_shape[i] = 1
       del y_shape[x.batch_dim_axis]
     else:
       if out_batch_dim_axis in axes:
         out_batch_dim_axis = None
+      if out_time_dim_axis in axes:
+        out_time_dim_axis = None
+      if out_feature_dim_axis is not NotSpecified and out_feature_dim_axis in axes:
```

Comment:
You don't need the `out_feature_dim_axis is not NotSpecified` here.

Reply:
fixed